dynamic_rest/renderers.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from rest_framework import renderers
+
+
+class DynamicJSONRenderer(renderers.JSONRenderer):
+
+  def render(self, data, accepted_media_type=None, renderer_context=None):
+    # if data is a string, just render it out using the default method
+    # this usually indicates an error message
+    if isinstance(data, basestring):
+      return super(DynamicJSONRenderer, self).render(data, accepted_media_type, renderer_context)
+
+    # populated by _sideload
+    self._data = {}
+
+    # used by _sideload to prevent duplicates
+    self._seen = defaultdict(set)
+
+    # used by _sideload to prevent adding secondary records of the primary resource
+    # from ending up in the primary slot
+    self._plural_name = getattr(
+        renderer_context.get('view').get_serializer().Meta, 'plural_name', 'objects')
```

Comment:
That makes sense -- I'll add `_get_plural_name` -- going to stick with the convention of prefixing our extension methods with `_` to maintain a clear distinction between DRF and our add-ons.